### DISCUSSION TITLE
receiving emails

### DISCUSSION THREAD

**Discussion by uPagge** [quarkusio/quarkus#28275] [Q&A]
Hello
For the project, we need the ability to receive letters, but in the documentation I found only a page about sending letters and nothing at all about receiving? Can anyone point my search in the right direction?

**Comment by rsvoboda** [comment]
Mailer extension handles sending of the emails. This extension doesn't cover functionality for receiving emails. 
One option could be to use Jakarta Mail (https://eclipse-ee4j.github.io/mail/) for fetching emails. CC @cescoffier

**Reply by cescoffier** [reply]
Exactly, we do not have an API to receive an email at the moment. Email reception can be unreliable, so, I would recommend using a message queue instead (which would store the received emails). 

About Jakarta Mail, be careful that it is a blocking API and may not work in native without a bit of work.

**Reply by eya99** [reply]
Hello i have a question do you have idea how can i use quarkus with azure auth to receiving emails

**Comment by uPagge** [comment]
Even sending has some issues, as we want users to be able to link mailboxes to their accounts. So far, it is also not clear how to make sure that each user has the opportunity to send a message from his mailbox. But we will figure it out, I will be glad if you direct me to useful materials))

**Reply by cescoffier** [reply]
You should be able to instantiate a MailClient for each users. However, be aware of the security implications of this approach. 

What about setting the 'from' or reply-to fields from the sent emails?

**Comment by humcqc** [comment]
Hello,

Is there some news about mail monitor in quarkus ? Is there some plans ?
What about camel-mail extension ?

I would need this kind of features, if nothing is available any advise to start some prototype ?

Thanks

**Reply by cescoffier** [reply]
Yes, you can use camel for this.
